nx/include/switch/services/capsa.h

```diff
@@ -18,31 +18,268 @@ void capsaExit(void);
+ * @param[out] width Output image width. Optional, can be NULL.
+ * @param[out] height Output image height. Optional, can be NULL.
+ * @param[in] file_id \ref CapsAlbumFileId
+ * @param[in] opts \ref CapsScreenShotDecodeOption
+ * @param[out] image RGBA8 image output buffer.
+ * @param[in] image_size Image buffer size, should be at least large enough for RGBA8 320x180.
+ * @param[out] workbuf Work buffer, cleared to 0 by the cmd before it returns.
+ * @param[in] workbuf_size Work buffer size, must be at least the size of the JPEG within the AlbumFile.
+ */
+Result capsaLoadAlbumScreenShotThumbnailImageEx(u64* width, u64* height, const CapsAlbumFileId *file_id, const CapsScreenShotDecodeOption *opts, void* image, u64 image_size, void* workbuf, u64 workbuf_size);
+
+/**
+ * @brief Returns the AlbumUsage for a specified \ref CapsAlbumStorage.
+ * @note Only available on [4.0.0+].
+ * @param[in] storage \ref CapsAlbumStorage
+ * @param[out] out \ref CapsAlbumUsage
+ */
 Result capsaGetAlbumUsage3(CapsAlbumStorage storage, CapsAlbumUsage3 *out);
+
+/**
+ * @brief Returns the result for a AlbumStorage mount.
+ * @param[in] storage \ref CapsAlbumStorage
+ */
 Result capsaGetAlbumMountResult(CapsAlbumStorage storage);
+
+/**
+ * @brief Returns the AlbumUsage for a specified \ref CapsAlbumStorage.
+ * @note Only available on [4.0.0+].
+ * @param[in] storage \ref CapsAlbumStorage
+ * @param[out] out \ref CapsAlbumUsage
+ */
 Result capsaGetAlbumUsage16(CapsAlbumStorage storage, CapsAlbumUsage16 *out);
 
+/**
+ * @brief Gets the currently set autosaving storage.
+ * @note Wrapper around setsysGetPrimaryAlbumStorage but defaults to NAND if SD isn't available.
+ * @param[out] storage \ref CapsAlbumStorage
+ */
 Result capsaGetAutoSavingStorage(CapsAlbumStorage* storage);
+
+/**
+ * @brief Gets required size to copy all files from one Storage to another.
+ * @param[in] dst_storage \ref CapsAlbumStorage
+ * @param[in] src_storage \ref CapsAlbumStorage
+ * @param[out] out Required storage space size.
+ */
 Result capsaGetRequiredStorageSpaceSizeToCopyAll(CapsAlbumStorage dst_storage, CapsAlbumStorage src_storage, u64* out);
-/// Only available on [4.0.0+].
-Result capsaLoadAlbumScreenShotThumbnailImageEx1(const CapsAlbumFileId *file_id, const CapsScreenShotDecodeOption *opts, void* workbuf, u64 workbuf_size, void* rawbuf, u64 rawbuf_size, void* out, u64 out_size);
 
+/**
+ * @brief Load the ScreenShotThumbnailImage for the specified AlbumFile.
+ * @note Only available on [4.0.0+].
+ * @param[in] file_id \ref CapsAlbumFileId
+ * @param[in] opts \ref CapsScreenShotDecodeOption
+ * @param[out] image RGBA8 image output buffer.
+ * @param[in] image_size Image buffer size, should be at least large enough for RGBA8 320x180.
+ * @param[out] workbuf Work buffer, cleared to 0 by the cmd before it returns.
+ * @param[in] workbuf_size Work buffer size, must be at least the size of the JPEG within the AlbumFile.
+ * @param[out] out Data buffer.
```

Comment:
Clarify what this is?

Reply:
I don't know ^^

Reply:
I'll check if I can find something when I get home.

Reply:
Probably the same UserData used by capsu, verify that?

Reply:
Checked and it's width, height, CapsScreenShotAttribute and then it all zeroes.
If it is CapsLoadAlbumScreenShotImageOutputForApplication it's userdata is empty.

Reply:
It's a fixed-size buffer, so replace out/out_size with {ptr for relevant struct}.

Reply:
Shall I remove "ForApplication" in this and "CapsScreenShotAttributeForApplication" as it isn't exclusively for Applications?

Reply:
Add 0x450-byte struct `CapsLoadAlbumScreenShotImageOutput` and use that.